internal/sync/secret_sync.go

```diff
@@ -94,14 +93,14 @@ func (s *SecretSync) SyncObjects() {
 	allSet := true
 
 	for k, v := range s.DefaultSynchronizer.Source.Status.Variables {
-		if b64value, found := s.Secret.Data[k]; !found || base64.StdEncoding.EncodeToString([]byte(v)) != string(b64value) {
+		if value, found := s.Secret.Data[k]; !found || v != string(value) {
 			allSet = false
 			break
 		}
 	}
 
 	if allSet {
-		s.Secret.StringData = map[string]string{}
+		s.Destination = nil
```

Comment:
This object is never nil as it comes from template, in this occasion we simply prevent patching the secret due to issues with resource version bump in SSA, even if stringData field is empty.

https://github.com/rancher/turtles/pull/916/files#diff-a7aa96fd7dfac77fd8444f38fabfa66c12a4b81ee2987baf5f1be4a39d74c917R64

Reply:
I mean this will probably panic: https://github.com/rancher/turtles/blob/main/internal/sync/core.go#L53
But then I see no tests failing in the PR, so I'm not sure anymore. Is this case covered?

Reply:
We are getting the resource before performing sync: https://github.com/rancher/turtles/blob/ff062aa851c33479aad2ca09e1f8f1ccb49455f9/internal/sync/interface.go#L52
and the destination is populated via template on initialization in the default synchronizer every other inherits from: https://github.com/rancher/turtles/blob/ff062aa851c33479aad2ca09e1f8f1ccb49455f9/internal/sync/provider_sync.go#L51-L61

Since we were using SSA patch, there was no need get the resource initially, but unfortunately the `secret` sync requires to check the content of the secret first, we could just `Apply` it… But it comes before the user-exposed `Sync` interface, so nothing overrides it with nil.

Original setting nullifying `stringData` was causing resourceVersion bumps regardless, so this `nil` check is meant to skip the `Apply`. I can add some tests covering the flow

Reply:
I believe this should be resolved now, it seems to be an issue with the usage of the SSA patch field owner. We were patching same resource with 2 different patches in the same reconcile loop using same owner, which caused infinite RV bumps on the provider secrets.

Existing nil check is covered via tests, as well as the flow of synchronizer. We shouldn’t get resources after we sync, the sync is to apply the resources from the template. I don’t see an issue requiring a change in the flow/order which would warrant this concern.